### ISSUE TITLE
Why use UserAgentMiddleware when DefaultHeadersMiddleware exists?

### ISSUE THREAD

**Issue by hduyyg** [scrapy/scrapy#3083]

**Comment by kmike** [comment]
Right, I don't see a good reason to have both. Maybe it is there for backwards compatibility. There is an extra (undocumented?) feature in UserAgentMiddleware though - it allows to change user-agent using spider's .useragent attribute.

**Comment by hduyyg** [comment]
The DefaultHeadersMiddleware can change  User-agent using spider's .headers attribute.
'I believe it's just for the reason of simplicity' - this is an answer from StackOverflow.
Maybe it's right?

**Comment by wRAR** [comment]
If `UserAgentMiddleware` is removed, will the users need to replace `USER_AGENT` with `DEFAULT_REQUEST_HEADERS`? That's very backwards-incompatible and I think inconvenient. It would make sense to not have `UserAgentMiddleware` from the beginning but as it exists I don't see how can it be removed (I guess `DefaultHeadersMiddleware` could handle `USER_AGENT` explicitly but I don't know if making this change is worth it).
